smilerating/src/main/java/com/hsalf/smilerating/SmileRating.java

```diff
@@ -311,31 +317,33 @@ protected void onDraw(Canvas canvas) {
-                mBackgroundPaint.setColor((Integer) mColorEvaluator
-                        .evaluate(mMainSmileyTransformaFraction, mPlaceHolderCirclePaint.getColor(),
-                                (mSelectedSmile == TERRIBLE || mPreviousSmile == TERRIBLE) ? mAngryColor : mNormalColor));
-                mScaleMatrix.reset();
-                mSmilePath.computeBounds(mScaleRect, true);
-                float nonSelectedScale = mFloatEvaluator.evaluate(
-                        mInterpolator.getInterpolation(mMainSmileyTransformaFraction), getScale(NONE), 1f);
-                mScaleMatrix.setScale(nonSelectedScale, nonSelectedScale,
-                        mScaleRect.centerX(), mScaleRect.centerY());
-                mDummyDrawPah.reset();
-                mDummyDrawPah.addPath(mSmilePath, mScaleMatrix);
-
-                canvas.drawCircle(mFaceCenter.x, mFaceCenter.y,
-                        nonSelectedScale * (mHeight / 2f), mBackgroundPaint);
-                canvas.drawPath(mDummyDrawPah, mPathPaint);
-            } else {
-                canvas.drawCircle(mFaceCenter.x, mFaceCenter.y, mHeight / 2f, mBackgroundPaint);
-                canvas.drawPath(mSmilePath, mPathPaint);
-            }
+//            if (mSmileyNotSelectedPreviously) {
+//                Log.i(TAG, "Non selection");
+//                /*mPathPaint.setAlpha(Math.round(255 * mMainSmileyTransformaFraction));
+//                mBackgroundPaint.setAlpha(Math.round(255 * mMainSmileyTransformaFraction));*/
+//                mPathPaint.setColor((Integer) mColorEvaluator
+//                        .evaluate(mMainSmileyTransformaFraction, mPlaceHolderFacePaint.getColor(), mDrawingColor));
+//                mBackgroundPaint.setColor((Integer) mColorEvaluator
+//                        .evaluate(mMainSmileyTransformaFraction, mPlaceHolderCirclePaint.getColor(),
+//                                (mSelectedSmile == TERRIBLE || mPreviousSmile == TERRIBLE) ? mAngryColor : mNormalColor));
+//                mScaleMatrix.reset();
+//                mSmilePath.computeBounds(mScaleRect, true);
+//                float nonSelectedScale = mFloatEvaluator.evaluate(
+//                        mInterpolator.getInterpolation(mMainSmileyTransformaFraction), getScale(NONE), 1f);
+//                mScaleMatrix.setScale(nonSelectedScale, nonSelectedScale,
+//                        mScaleRect.centerX(), mScaleRect.centerY());
+//                mDummyDrawPah.reset();
+//                mDummyDrawPah.addPath(mSmilePath, mScaleMatrix);
+//
+//                canvas.drawCircle(mFaceCenter.x, mFaceCenter.y,
+//                        nonSelectedScale * (mHeight / 2f), mBackgroundPaint);
+//                canvas.drawPath(mDummyDrawPah, mPathPaint);
+//            } else {
+//                canvas.drawCircle(mFaceCenter.x, mFaceCenter.y, mHeight / 2f, mBackgroundPaint);
+//                canvas.drawPath(mSmilePath, mPathPaint);
+//            }
```

Comment:
I commented out this code block because there is a big problem inside this. Let Me explain you

1. Let's ignore my changes for this commented code-block only. Revert back this commented code block to the original one and run the app using that code.
2. Now tap on any smiley (first time) and then release it and after that try to swipe over the smilies you will see that the smilies colors aren't changing while swiping. And When you tap on any smiley (second time) and then swipe over the smilies you will see that the smilies colors are changing.

Basically, in short On First Tap and then Swipe over smilies the color change is not working for independent smilies. Even in the case of current stable master branch